### PR TITLE
Fixed issue #T1055: Optimization open/closed-access mode and survey activation screens

### PR DESCRIPTION
**Pull request by thedirtypanda**

### CODE REVIEW

File: application/controllers/admin/tokens.php [hunk 1]
```diff
@@ -2499,7 +2499,7 @@ public function _handletokenform($iSurveyId, $subaction, $iTokenId = "", $ajax =
      * @param int $iSurveyId
      * @return void
      */
-    private function _newtokentable($iSurveyId)
+    private function _newtokentable(int $iSurveyId)
```
Comment (gabrieljenik) [inline application/controllers/admin/tokens.php:2502]: Is the typhint needed?
Are we using this for other method signatures?

**Review by gabrieljenik — APPROVED** [review]
Code OK.
Not tested

**Comment by glimz** [comment]
@thedirtypanda can you fix the merge conflict? So I can merge your fix?

**Comment by thedirtypanda** [comment]
@glimz Yeah wait.

File: application/controllers/admin/tokens.php (hunk 1 above)
Comment (Shnoulle) [reply application/controllers/admin/tokens.php]: And it's great (because it send 400 error)

**Pull request merged**